### DISCUSSION TITLE
unrecognized GRIB1 field abbreviation 'APCP_24'

### DISCUSSION THREAD

**Discussion by JosemariePalero** [dtcenter/METplus#2640] [Configuration] [answered]
[Update: I tried to change the variable name of the netCDF file from APCP_24 to APCP and I get the following error:
point_stat ${METPLUS_DATA}/fcst/WRF/processed/2021/12/WRFrr_24h20211201.nc ${METPLUS_DATA}/obs/synop/processed/nc/rr_24h/2021/12/synoprr_24h202112010000.nc /home/erdanx/ai4rp/user_config/PointStatConfig -outdir /home/erdanx/ai4rp/output/pointStat -v 5
DEBUG 1: Start point_stat by erdanx(1000) at 2024-07-19 05:35:36Z  cmd: point_stat /home/erdanx/ai4rp/data/fcst/WRF/processed/2021/12/WRFrr_24h20211201.nc /home/erdanx/ai4rp/data/obs/synop/processed/nc/rr_24h/2021/12/synoprr_24h202112010000.nc /home/erdanx/ai4rp/user_config/PointStatConfig -outdir /home/erdanx/ai4rp/output/pointStat -v 5 
DEBUG 1: Default Config File: /home/erdanx/WRF/MET-11.1.1/share/met/config/PointStatConfig_default
DEBUG 1: User Config File: /home/erdanx/ai4rp/user_config/PointStatConfig
DEBUG 4: Met2dDataFileFactory::new_met_2d_data_file() -> created new Met2dDataFile object of type "FileType_None".
ERROR  : 
ERROR  : Trouble reading forecast file "/home/erdanx/ai4rp/data/fcst/WRF/processed/2021/12/WRFrr_24h20211201.nc"
ERROR  : 

The netCDF file I'm using is an output of the PCP Combine tool

Now, I used the CF Compliance Checker and here are the details:
Uploaded File: WRFrr_24h20211201.nc

CHECKING NetCDF FILE: /tmp/tmpz0ixo91o.nc
=====================
Using CF Checker Version 4.1.0
Checking against CF Version CF-1.8
Using Standard Name Table Version 85 (2024-05-21T15:55:10Z)
Using Area Type Table Version 11 (06 July 2023)
Using Standardized Region Name Table Version 4 (18 December 2018)

WARN: (2.6.1): No 'Conventions' attribute present

------------------
Checking variable: lat
------------------

------------------
Checking variable: lon
------------------

------------------
Checking variable: APCP
------------------

ERRORS detected: 0
WARNINGS given: 1
INFORMATION messages: 0

----
]

Hello I am currently working on verifying the WRF output using point synoptic data using the Point Stat of the METplus. When I am trying to use Point Stat, I get the following error:

(ai4rp) erdanx@erdanx:~/ai4rp$ point_stat ${METPLUS_DATA}/fcst/WRFrr24h.nc ${METPLUS_DATA}/obs/synoprr24hFmtd/nc/2021/synoprr24h202112010000.nc /home/erdanx/ai4rp/user_config/PointStatConfig -outdir /home/erdanx/ai4rp/output/pointStat -v 2
DEBUG 1: Start point_stat by erdanx(1000) at 2024-07-18 23:53:07Z  cmd: point_stat /home/erdanx/ai4rp/data/fcst/WRFrr24h.nc /home/erdanx/ai4rp/data/obs/synoprr24hFmtd/nc/2021/synoprr24h202112010000.nc /home/erdanx/ai4rp/user_config/PointStatConfig -outdir /home/erdanx/ai4rp/output/pointStat -v 2 
DEBUG 1: Default Config File: /home/erdanx/WRF/MET-11.1.1/share/met/config/PointStatConfig_default
DEBUG 1: User Config File: /home/erdanx/ai4rp/user_config/PointStatConfig
GSL_RNG_TYPE=mt19937
GSL_RNG_SEED=18446744073234192742
DEBUG 1: Forecast File: /home/erdanx/ai4rp/data/fcst/WRFrr24h.nc
DEBUG 1: Observation File: /home/erdanx/ai4rp/data/obs/synoprr24hFmtd/nc/2021/synoprr24h202112010000.nc
DEBUG 2: 
DEBUG 2: --------------------------------------------------------------------------------
DEBUG 2: 
DEBUG 2: Reading data for APCP_24/A24.
DEBUG 2: Processing masking regions.
DEBUG 2: Processing geography data.
DEBUG 2: For APCP_24/A24 found 1 forecast levels, 0 climatology mean levels, and 0 climatology standard deviation levels.
DEBUG 2: 
DEBUG 2: --------------------------------------------------------------------------------
DEBUG 2: 
ERROR  : 
ERROR  : VarInfoGrib::add_grib_code() -> unrecognized GRIB1 field abbreviation 'APCP_24' for table version (-9999), center (-9999), and subcenter (-9999) or default table version (2), center (7), and subcenter (1).
ERROR  : 

I have used the PCP Combine for my WRF data and I have used the ASCII2NC tool for my point observation data. 

Here are some information for my WRF data:
{'lat': <class 'netCDF4._netCDF4.Variable'>
float32 lat(lat, lon)
    long_name: latitude
    units: degrees_north
    standard_name: latitude
unlimited dimensions: 
current shape = (213, 181)
filling on, default _FillValue of 9.969209968386869e+36 used, 'lon': <class 'netCDF4._netCDF4.Variable'>
float32 lon(lat, lon)
    long_name: longitude
    units: degrees_east
    standard_name: longitude
unlimited dimensions: 
current shape = (213, 181)
filling on, default _FillValue of 9.969209968386869e+36 used, 'APCP_24': <class 'netCDF4._netCDF4.Variable'>
float32 APCP_24(lat, lon)
    coordinates: lat lon
    name: APCP_24
    long_name: Total Precipitation
    level: A24
    units: kg/m^2
    _FillValue: -9999.0
    init_time: 20211201_000000
    init_time_ut: 1638316800
    valid_time: 20211202_000000
    valid_time_ut: 1638403200
    accum_time: 240000
    accum_time_sec: 86400
unlimited dimensions: 
current shape = (213, 181)
filling on}

And here are some information for my point observation data:
{'hdr_typ': <class 'netCDF4._netCDF4.Variable'>
int32 hdr_typ(nhdr)
    long_name: index of message type
    _FillValue: -9999
unlimited dimensions: 
current shape = (55,)
filling on, 'hdr_sid': <class 'netCDF4._netCDF4.Variable'>
int32 hdr_sid(nhdr)
    long_name: index of station identification
    _FillValue: -9999
unlimited dimensions: 
current shape = (55,)
filling on, 'hdr_vld': <class 'netCDF4._netCDF4.Variable'>
int32 hdr_vld(nhdr)
    long_name: index of valid time
    _FillValue: -9999
unlimited dimensions: 
current shape = (55,)
filling on, 'hdr_lat': <class 'netCDF4._netCDF4.Variable'>
float32 hdr_lat(nhdr)
    long_name: latitude
    _FillValue: -9999.0
    units: degrees_north
unlimited dimensions: 
current shape = (55,)
filling on, 'hdr_lon': <class 'netCDF4._netCDF4.Variable'>
float32 hdr_lon(nhdr)
    long_name: longitude
    _FillValue: -9999.0
    units: degrees_east
unlimited dimensions: 
current shape = (55,)
filling on, 'hdr_elv': <class 'netCDF4._netCDF4.Variable'>
float32 hdr_elv(nhdr)
    long_name: elevation
    _FillValue: -9999.0
    units: meters above sea level (msl)
unlimited dimensions: 
current shape = (55,)
filling on, 'obs_qty': <class 'netCDF4._netCDF4.Variable'>
int32 obs_qty(nobs)
    long_name: index of quality flag
unlimited dimensions: 
current shape = (55,)
filling on, default _FillValue of -2147483647 used, 'obs_hid': <class 'netCDF4._netCDF4.Variable'>
int32 obs_hid(nobs)
    long_name: index of matching header data
    _FillValue: -9999
unlimited dimensions: 
current shape = (55,)
filling on, 'obs_gc': <class 'netCDF4._netCDF4.Variable'>
int32 obs_gc(nobs)
    long_name: grib code corresponding to the observation type
    _FillValue: -9999
unlimited dimensions: 
current shape = (55,)
filling on, 'obs_lvl': <class 'netCDF4._netCDF4.Variable'>
float32 obs_lvl(nobs)
    long_name: pressure level (hPa) or accumulation interval (sec)
    _FillValue: -9999.0
unlimited dimensions: 
current shape = (55,)
filling on, 'obs_hgt': <class 'netCDF4._netCDF4.Variable'>
float32 obs_hgt(nobs)
    long_name: height in meters above sea level or ground level (msl or agl)
    _FillValue: -9999.0
unlimited dimensions: 
current shape = (55,)
filling on, 'obs_val': <class 'netCDF4._netCDF4.Variable'>
float32 obs_val(nobs)
    long_name: observation value
    _FillValue: -9999.0
unlimited dimensions: 
current shape = (55,)
filling on, 'hdr_typ_table': <class 'netCDF4._netCDF4.Variable'>
|S1 hdr_typ_table(nhdr_typ, mxstr2)
    long_name: message type
unlimited dimensions: 
current shape = (1, 40)
filling on, default _FillValue of � used, 'hdr_sid_table': <class 'netCDF4._netCDF4.Variable'>
|S1 hdr_sid_table(nhdr_sid, mxstr2)
    long_name: station identification
unlimited dimensions: 
current shape = (55, 40)
filling on, default _FillValue of � used, 'hdr_vld_table': <class 'netCDF4._netCDF4.Variable'>
|S1 hdr_vld_table(nhdr_vld, mxstr)
    long_name: valid time
    units: YYYYMMDD_HHMMSS UTC
unlimited dimensions: 
current shape = (1, 16)
filling on, default _FillValue of � used, 'obs_qty_table': <class 'netCDF4._netCDF4.Variable'>
|S1 obs_qty_table(nobs_qty, mxstr)
    long_name: quality flag
unlimited dimensions: 
current shape = (1, 16)
filling on, default _FillValue of � used}

Here are some information about my Point Stat config:
////////////////////////////////////////////////////////////////////////////////
//
// Point-Stat configuration file.
//
// For additional information, please see the MET User's Guide.
//
////////////////////////////////////////////////////////////////////////////////

//
// Output model name to be written
//
model = "WRF";

//
// Output description to be written
// May be set separately in each "obs.field" entry
//
desc = "NA";

////////////////////////////////////////////////////////////////////////////////

//
// Verification grid
// May be set separately in each "field" entry
//
regrid = {
   to_grid    = NONE;
   method     = NEAREST;
   width      = 1;
   vld_thresh = 0.5;
   shape      = SQUARE;
}

////////////////////////////////////////////////////////////////////////////////

//
// May be set separately in each "field" entry
//
censor_thresh  = [];
censor_val     = [];
mpr_column     = [];
mpr_thresh     = [];
cat_thresh     = [ NA ];
cnt_thresh     = [ NA ];
cnt_logic      = UNION;
wind_thresh    = [ NA ];
wind_logic     = UNION;
eclv_points    = 0.05;
hss_ec_value   = NA;
rank_corr_flag = FALSE;

//
// Forecast and observation fields to be verified
//
fcst = {
   field = [
      {
        name       = "APCP_24";
        level      = [ "A24" ];
        ctc_thresh = [ <=1, <=5, <=10, <=20, <=50, <=100 ];
        cts_thresh = [ <=1, <=5, <=10, <=20, <=50, <=100 ];
      }
   ];

}
obs = fcst;

////////////////////////////////////////////////////////////////////////////////

//
// Point observation filtering options
// May be set separately in each "obs.field" entry
//
message_type   = [ "ADPSFC" ];
sid_inc        = [];
sid_exc        = [];
obs_quality_inc  = [];
obs_quality_exc  = [];
duplicate_flag = NONE;
obs_summary    = NONE;
obs_perc_value = 50;

//
// Mapping of message type group name to comma-separated list of values.
//
message_type_group_map = [
   { key = "SURFACE"; val = "ADPSFC,SFCSHP,MSONET";               },
   { key = "ANYAIR";  val = "AIRCAR,AIRCFT";                      },
   { key = "ANYSFC";  val = "ADPSFC,SFCSHP,ADPUPA,PROFLR,MSONET"; },
   { key = "ONLYSF";  val = "ADPSFC,SFCSHP";                      },
   { key = "LANDSF";  val = "ADPSFC,MSONET";                      },
   { key = "WATERSF"; val = "SFCSHP";                             }
];

////////////////////////////////////////////////////////////////////////////////

//
// Climatology data
//
climo_mean = {

   file_name = [];
   field     = [];

   regrid = {
      method     = NEAREST;
      width      = 1;
      vld_thresh = 0.5;
      shape      = SQUARE;
   }

   time_interp_method = DW_MEAN;
   day_interval       = 31;
   hour_interval      = 6;
}

climo_stdev = climo_mean;
climo_stdev = {
   file_name = [];
}

//
// May be set separately in each "obs.field" entry
//
climo_cdf = {
   cdf_bins    = 1;
   center_bins = FALSE;
   write_bins  = TRUE;
   direct_prob = FALSE;
}

////////////////////////////////////////////////////////////////////////////////

//
// Land/Sea mask
// For LANDSF message types, only use forecast grid points where land = TRUE.
// For WATERSF message types, only use forecast grid points where land = FALSE.
// land_mask.flag may be set separately in each "obs.field" entry.
//
land_mask = {
   flag      = FALSE;
   file_name = [];
   field     = { name = "LAND"; level = "L0"; }
   regrid    = { method = NEAREST; width = 1; }
   thresh    = eq1;
}

//
// Topography
// For SURFACE message types, only use observations where the topo - station
// elevation difference meets the use_obs_thresh threshold.
// For the observations kept, when interpolating forecast data to the
// observation location, only use forecast grid points where the topo - station
// difference meets the interp_fcst_thresh threshold.
// topo_mask.flag may be set separately in each "obs.field" entry.
//
topo_mask = {
   flag               = FALSE;
   file_name          = [];
   field              = { name = "TOPO"; level = "L0"; }
   regrid             = { method = BILIN; width = 2; }
   use_obs_thresh     = ge-100&&le100;
   interp_fcst_thresh = ge-50&&le50;
}

////////////////////////////////////////////////////////////////////////////////

//
// Point observation time window
// May be set separately in each "obs.field" entry
//
obs_window = {
   beg = -5400;
   end =  5400;
}

////////////////////////////////////////////////////////////////////////////////

//
// Verification masking regions
// May be set separately in each "obs.field" entry
//
mask = {
   grid  = [ "FULL" ];
   poly  = [];
   sid   = [];
   llpnt = [];
}

////////////////////////////////////////////////////////////////////////////////

//
// Confidence interval settings
// May be set separately in each "obs.field" entry
//
ci_alpha  = [ 0.05, 0.10 ];

boot = {
   interval = PCTILE;
   rep_prop = 1.0;
   n_rep    = 0;
   rng      = "mt19937";
   seed     = "";
}

////////////////////////////////////////////////////////////////////////////////

//
// Interpolation methods
// May be set separately in each "obs.field" entry
//
interp = {
   vld_thresh = 1.0;
   shape      = SQUARE;

   type = [
      {
         method = NEAREST;
         width  = 1;
      }
   ];
}

////////////////////////////////////////////////////////////////////////////////

//
// HiRA verification method
// May be set separately in each "obs.field" entry
//
hira = {
   flag            = FALSE;
   width           = [ 2, 3, 4, 5 ];
   vld_thresh      = 1.0;
   cov_thresh      = [ ==0.25 ];
   shape           = SQUARE;
   prob_cat_thresh = [];
}

////////////////////////////////////////////////////////////////////////////////

//
// Threshold for SEEPS p1 (Probability of being dry)
//
seeps_p1_thresh = >=0.1&&<=0.85;

////////////////////////////////////////////////////////////////////////////////

//
// Statistical output types
// May be set separately in each "obs.field" entry
//
output_flag = {
   fho    = NONE;
   ctc    = BOTH;
   cts    = BOTH;
   mctc   = NONE;
   mcts   = NONE;
   cnt    = NONE;
   sl1l2  = NONE;
   sal1l2 = NONE;
   vl1l2  = NONE;
   val1l2 = NONE;
   vcnt   = NONE;
   pct    = NONE;
   pstd   = NONE;
   pjc    = NONE;
   prc    = NONE;
   ecnt   = NONE; // Only for HiRA.
   orank  = NONE; // Only for HiRA.
   rps    = NONE; // Only for HiRA.
   eclv   = NONE;
   mpr    = NONE;
   seeps  = NONE;
   seeps_mpr = NONE;
}

////////////////////////////////////////////////////////////////////////////////

tmp_dir        = "/tmp";
output_prefix  = "pointStat";
version        = "V11.1.0";

////////////////////////////////////////////////////////////////////////////////

**Comment by j-opatz** [comment, marked as answer]
Hello,

And thank you for your question. I think the fix for this problem may be in how you are calling the forecast field level.

As you stated, you've already successfully used the PCP-Combine tool on your WRF file; this means that the input file is a netCDF. But in your forecast dictionary's level setting you're using 
```
fcst = {
field = [
{
name = "APCP_24";
level = [ "A24" ];
```
When using a netCDF file, you must use the proper level notation for the file type. For netCDFs, this is parentheses, along with the indicies for each array of data. Since APCP_24 is defined by `(lat,lon)`, the correct notation would be
```
fcst = {
field = [
{
name = "APCP_24";
level = [ "(*,*)" ];
```
Note that the asterisks are used to convey to MET that you would like all values associated with each array of data (as opposed to an individual row, column, or value).  You can read more about the netCDF settings available from the forecast or observation dictionaries in the [MET User's Guide](https://met.readthedocs.io/en/latest/Users_Guide/config_options.html#fcst) (scroll down from this link to the "NetCDF (from MET tools, CF-compliant, p_interp, and wrf_interp):" bullet).

I'd also caution that as you currently have your configuration file set, your observation dictionary values (variable name, levels, thresholding, etc.) will be a copy of your forecast dictionary values, but you are passing in a different observation file from your forecast file:
```
obs = fcst;
```

You will need to create a observation dictionary that has its own field information, including the appropriate variable name and level information, that matches the contents of your ASCII2NC output file.

**Comment by j-opatz** [comment]
Hello,

I wanted to reach out again to see if the suggestion helped you through your issue. If it has, please select the comment as an answer and I'll go ahead and lock the Discussion.

**Reply by JosemariePalero** [reply]
Hi, 

I apologize for the late reply. I was on vacation last week, so I was not able to give feedback. Your suggestion really helped me through my problem. Thank you very much for your guidance.
